src/content/app/tools/blast/components/blast-input-sequences/useBlastInputSequences.test.tsx

```diff
@@ -0,0 +1,254 @@
+          sequences: [{ header: 'foo', value: 'MENLNMDL' }],
+          settings: {
+            ...initialBlastFormState.settings,
+            sequenceSelectionMode: 'manual',
+            sequenceType: 'protein'
+          }
+        }
+      });
+      const { result: result2 } = renderHook(() => useBlastInputSequences(), {
+        wrapper: wrapper2
+      });
+      updateSequences = result2.current.updateSequences;
+
+      updateSequences([]);
+
+      expect(getSelectedSequenceType(store2.getState() as any)).toEqual('dna'); // sequence type reset to initial
+    });
+
+    it('does not change sequence type if user has changed sequence type manually', () => {
+      const proteinSequence =
+        'MENLNMDLLYMAAAVMMGLAAIGAAIGIGILGGKFLEGAARQPDLIPLLRTQFFIVMGLVDAIPMIAVGL';
+      const { wrapper, store } = getWrapper({
+        state: {
+          settings: {
+            ...initialBlastFormState.settings,
+            sequenceSelectionMode: 'manual'
+          }
+        }
+      });
+      const { result } = renderHook(() => useBlastInputSequences(), {
+        wrapper
+      });
+      const { updateSequences } = result.current;
+      expect(getSelectedSequenceType(store.getState() as any)).toEqual('dna'); // initial value
+
+      updateSequences([{ value: proteinSequence }]);
+
+      expect(getSelectedSequenceType(store.getState() as any)).toEqual('dna'); // sequence type should not have changed
+    });
+  });
+
+  describe('updateSequenceType', () => {
+    it('changes sequence type and sets the change type to manual', () => {
+      const { wrapper, store } = getWrapper();
+      const { result } = renderHook(() => useBlastInputSequences(), {
+        wrapper
+      });
+
+      // initial values
+      expect(getSelectedSequenceType(store.getState() as any)).toEqual('dna');
+      expect(getSequenceSelectionMode(store.getState() as any)).toEqual(
+        'automatic'
+      );
+
+      const { updateSequenceType } = result.current;
+      updateSequenceType('protein');
+
+      expect(getSelectedSequenceType(store.getState() as any)).toEqual(
+        'protein'
+      );
+      expect(getSequenceSelectionMode(store.getState() as any)).toEqual(
+        'manual'
+      );
+    });
+  });
+
+  describe('clearAllSequences', () => {
+    it('clears sequences and sets the change type to automatic', () => {
+      const { wrapper, store } = getWrapper({
+        state: {
+          sequences: [{ header: 'foo', value: 'MENLNMDL' }],
+          settings: {
+            ...initialBlastFormState.settings,
+            sequenceSelectionMode: 'manual',
+            sequenceType: 'protein'
+          }
+        }
+      });
+      const { result } = renderHook(() => useBlastInputSequences(), {
+        wrapper
+      });
+
+      // initial values
+      expect(getSelectedSequenceType(store.getState() as any)).toEqual(
+        'protein'
+      );
+      expect(getSequenceSelectionMode(store.getState() as any)).toEqual(
+        'manual'
```

Comment:
Is it worth testing the initial state here?

Reply:
Not really; it was just a sanity check. It can go.

✅